### ISSUE TITLE
[stable/mariadb] There is a crash when the length of password is more than 32 bits

### ISSUE THREAD

**Issue by gwind** [helm/charts#16600]
When use a string lager than 32 bits password for `rootUser.password` , the ${release-name}-master-0 pod would be crashed.

**Reproducible steps** :

1. custom values.yaml ( set `rootUser.password` with a length more than 32 bits )
2. install a release

```shell
helm install --namespace my-production --name my-mariadb -f values.yaml ./mariadb
```
3. view the pod logs
```shell
➜  mariadb git:(master) ✗ kubectl -n my-production logs -f pod/my-mariadb-master-0 -c mariadb

Welcome to the Bitnami mariadb container
Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-mariadb
Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-mariadb/issues
Send us your feedback at containers@bitnami.com

INFO  ==> ** Starting MariaDB setup **
INFO  ==> Validating settings in MYSQL_*/MARIADB_* env vars..
/libmysql.sh: line 303: $1: unbound variable
```

The source code in `/libmysql.sh` is here:

```bash
270 ########################
271 # Validate settings in MYSQL_*/MARIADB_* environment variables
272 # Globals:
273 #   DB_*
274 # Arguments:
275 #   None
276 # Returns:
277 #   None
278 #########################
279 mysql_validate() {
280     info "Validating settings in MYSQL_*/MARIADB_* env vars.."
281
282     # Auxiliary functions
283     empty_password_enabled_warn() {
284         warn "You set the environment variable ALLOW_EMPTY_PASSWORD=${ALLOW_EMPTY_PASSWORD}. For safety reasons, do not use this flag in a production environment."
285     }
286     empty_password_error() {
287         error "The $1 environment variable is empty or not set. Set the environment variable ALLOW_EMPTY_PASSWORD=yes to allow the container to be started with blank passwords. This is recommended only for development."
288         exit 1
289     }
290
291     if [[ -n "$DB_REPLICATION_MODE" ]]; then
292         if [[ "$DB_REPLICATION_MODE" = "master" ]]; then
293             if is_boolean_yes "$ALLOW_EMPTY_PASSWORD"; then
294                 empty_password_enabled_warn
295             else
296                 if [[ -n "$DB_REPLICATION_USER" ]] && [[ -z "$DB_REPLICATION_PASSWORD" ]]; then
297                     empty_password_error "$(get_env_var REPLICATION_PASSWORD)"
298                 fi
299                 if [[ -z "$DB_ROOT_PASSWORD" ]]; then
300                     empty_password_error "$(get_env_var ROOT_PASSWORD)"
301                 fi
302                 if (( ${#DB_ROOT_PASSWORD} > 32 )); then
303                     error "The password can not be longer than 32 characters. Set the environment variable $1 with a shorter value"
304                     exit 1
305                 fi
306                 if [[ -n "$DB_USER" ]] && [[ -z "$DB_PASSWORD" ]]; then
307                     empty_password_error "$(get_env_var PASSWORD)"
308                 fi
309             fi
310         elif [[ "$DB_REPLICATION_MODE" = "slave" ]]; then
311             if [[ -z "$DB_MASTER_HOST" ]]; then
312                 error "Slave replication mode chosen without setting the environment variable $(get_env_var MASTER_HOST). Use it to indicate where the Master node is running"
313                 exit 1
314             fi
315         else
316             error "Invalid replication mode. Available options are 'master/slave'"
317             exit 1
318         fi
319     else
320         if is_boolean_yes "$ALLOW_EMPTY_PASSWORD"; then
321             empty_password_enabled_warn
322         else
323             if [[ -z "$DB_ROOT_PASSWORD" ]]; then
324                 empty_password_error "$(get_env_var ROOT_PASSWORD)"
325             fi
326             if [[ -n "$DB_USER" ]] && [[ -z "$DB_PASSWORD" ]]; then
327                 empty_password_error "$(get_env_var PASSWORD)"
328             fi
329         fi
330     fi
331 }
```

**Comment by carrodher** [comment]
Hi, thanks for reporting this issue.

It is true that the error is not very descriptive because a typo in the error message, I just created an internal task to fix it in future releases. In the file you mentioned, the change we should implement is something like:
```diff
-error "The password can not be longer than 32 characters. Set the environment variable $1 with a shorter value"
+error "The password can not be longer than 32 characters. Set the environment variable $(get_env_var ROOT_PASSWORD) with a shorter value (currently ${#DB_ROOT_PASSWORD} characters)"
```
With the above change, the container crash with the following message:
```bash
$ helm install --set image.tag=testing-changes --set rootUser.password=1111111111111111111111111111111111111111111111 --name my-mariadb -f values.yaml .
$ kubectl logs -f my-mariadb-master-0

Welcome to the Bitnami mariadb container
Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-mariadb
Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-mariadb/issues
Send us your feedback at containers@bitnami.com

INFO  ==> ** Starting MariaDB setup **
INFO  ==> Validating settings in MYSQL_*/MARIADB_* env vars..
ERROR ==> The password can not be longer than 32 characters. Set the environment variable MARIADB_ROOT_PASSWORD with a shorter value (currently 46 characters)
```

This message is valid, because according to the [changelog](https://dev.mysql.com/doc/relnotes/mysql/5.7/en/news-5-7-5.html), MariaDB/MySQL has some limitations in the number of characters when using replication:
> **Important Change**: Replication: The maximum length that can be used for the password in a `CHANGE MASTER TO` statement is 32 characters. Previously, when a longer password was employed, it was accepted, but any excess length was silently truncated by the server. Following this fix, when the password's length exceeds 32 characters, `CHANGE MASTER TO` fails with an error. (Bug #11752299, Bug #43439)

In summary, the container should stop the execution when the password is longer than 32 characters, but the error message that it was showing was not appropriate because of a typo. I just sent a PR fixing the error message, but users need to specify a password with less than 32 characters.

**Comment by carrodher** [comment]
Changes are already implemented in the image: https://github.com/bitnami/bitnami-docker-mariadb/commit/1f8917978ada4581f324ab8702d
I am going to bump the mariadb version used by the chart in order to use an image containing the latest changes.

**Comment by carrodher** [comment]
⬆️ stable/mariadb 6.8.3 should contain the new image with the descriptive error message. Thanks again for reporting the issue!
